### ISSUE TITLE
bug: custom completion with promise does trigger validation

### ISSUE THREAD

**Issue by buehler** [yargs/yargs#1235]
Hey there, I'm currently experiencing the following bug:

When using completions, a synchronuous handler does not trigger the validation (which is correct),
but when I have a command `foobar <arg>` and try to use the completion `node . --get-yargs-completions foobar` with a custom completion handler that returns a promise

```typescript
.completion('completion', false as any, (_, argv: Arguments) => {
        return Promise.resolve(['asdf']);
      }),
```

It does trigger the validation of the required argument and therefor executes the completion handler, but then the process gets exited with the error that a required argument is missing.

Also, when using promises, the handler of the command is executed as well.

**Comment by buehler** [comment]
As I investigated, the same happens when using `async`.

**Comment by bcoe** [comment]
@buehler not 100% sure when I fixed this, but it seems to be fixed in #1823 

```js
import yargs from 'yargs';
import { setTimeout } from 'timers/promises';

yargs(process.argv.slice(2))
  .command('foobar <arg>', 'the foobar command', (yargs) => {
    yargs.command('blerg', 'blerg command');
  })
  .completion('completion', false, async (_, argv) => {
    await setTimeout(1000)
    return Promise.resolve(['hello', 'world'])
  })
  .parse();
```

☝️ outputs:

```
hello
world
```

With no error.
